[chiayuxuan] iP

chiayuxuan opened the pull request.

chiayuxuan committed:
Add increment Level-0

chiayuxuan committed:
Add Level 0

chiayuxuan committed:
Add indentation for Level 0

chiayuxuan committed:
Add Level-1

chiayuxuan committed:
Add Level-2

chiayuxuan committed:
Add Tasks class

chiayuxuan committed:
Add Level-3

geraldkoh4 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,61 @@
+        Scanner in = new Scanner(System.in);
+       // ArrayList<Tasks> taskList = new ArrayList<>();
+        String line;
         String logo = " ____        _        \n"
```
1. Maybe the long print statements can be put into another class to look neater?

geraldkoh4 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,61 @@
+        System.out.print(divider + '\n'
+                + " \t Hello! I'm Duke\n\t What can I do for you?\n" + divider + "\n\n");
+        line = in.nextLine(); //first input
+        while (!line.equals("bye")) {
+            String[] userInput = line.split(" ", 2);
+            String command = userInput[0];
+            switch (command) {
+            case "mark":
```
2. I like the way this is presented. The switch - case helps a lot with readability!

geraldkoh4 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,61 @@
+            }
+            line = in.nextLine(); //new user input
+        }
+        System.out.println(divider + "\n\t Bye. Hope to see you again soon!\n" + divider);
```
3. Overall the code does not have deep nesting, good job!

geraldkoh4 commented on src/main/java/Tasks.java:
```diff
@@ -0,0 +1,52 @@
+    public void setMarked(boolean isMarked) {
+        this.isMarked = isMarked;
+    }
+}
```
4. Short method names help with readability. Looks neat

geraldkoh4 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,61 @@
+                        System.out.println("\t " + Integer.toString(num) + ".[ ] " + thisTask.getItem());
+                    }
+                    ++num;
+                }
```
5. Perhaps you could use a for loop to make it simpler? 
E.g for (int num = 1; num <= timer; num++) {
    //...
}

wenxin-c commented on src/main/java/Tasks.java:
```diff
@@ -0,0 +1,52 @@
+        this.id = numberOfTasks;
+    }
+
+    public static void addToList(Tasks task){
+        taskList.add(task);
+    }
+
+    public static ArrayList<Tasks> getTaskList() {
+        return taskList;
+    }
+
+    public Tasks getTask(int taskNum) {
+        return taskList.get(taskNum - 1); //zero-based
+    }
+
+    public static int getNumberOfTasks() {
+        return numberOfTasks;
+    }
+
+    public int getID() {
+        return id;
+    }
+
+    public String getItem() {
+        return item;
+    }
+
+    public void setItem(String item) {
+        this.item = item;
+    }
```
Good abstraction of getter and setter.

wenxin-c commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,61 @@
+        line = in.nextLine(); //first input
+        while (!line.equals("bye")) {
+            String[] userInput = line.split(" ", 2);
+            String command = userInput[0];
```
Good logic, but the code can be further abstracted into different classes, for example, a command manager class to manage commands instead of putting everything in main:D

wenxin-c reviewed (commented):
Overall good work!! Further abstractions can be added to improve the readability of the code:D

chiayuxuan committed:
Add some functionality classes: Dateline, CommandManager, Event, Todo

chiayuxuan committed:
Fixed Level 2 and 3

chiayuxuan committed:
Fixed bugs for Level 2 and 3

chiayuxuan committed:
Add functionality for todo

chiayuxuan committed:
Add Level-4

chiayuxuan committed:
Followed Coding Standards

chiayuxuan committed:
Added some exceptions

chiayuxuan committed:
Add exceptions and custom exception class

chiayuxuan committed:
Merge branch 'branch-Level-5'

R-Ramana commented on src/main/java/CommandManager.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    public String getUserInput() {
+
```
Unnecessary empty lines here and in `setKey(String key)` method, can remove them. Same issue noted in other files as well

chiayuxuan committed:
Changed Tasks to Abstract Class and fixed indentations

chiayuxuan committed:
Add delete function

chiayuxuan committed:
Added some small details

chiayuxuan committed:
Added functionality for Level-7

chiayuxuan committed:
Add Level-7

chiayuxuan committed:
Handled conflicts with 'branch-Level-6'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/Tasks.java
#	text-ui-test/runtest.bat

chiayuxuan committed:
Add Exception class and cleaned code

chiayuxuan committed:
Packaged classes

chiayuxuan committed:
Merge branch 'A-Packages'

chiayuxuan committed:
Added Storage class and refactored methods in Duke.java

chiayuxuan committed:
Add TaskList Class

chiayuxuan committed:
Merge branch 'TaskList'

chiayuxuan committed:
Add more OOP (WIP)

chiayuxuan committed:
Add exception methods and packaged classes

chiayuxuan committed:
Removed Commented out lines

chiayuxuan committed:
Removed obselete class

chiayuxuan committed:
Add FindCommand class for Level 9

chiayuxuan committed:
Created JAR file + Added more OOP + Some exception classes + Collections Framework

chiayuxuan committed:
Deleted empty spaces

chiayuxuan committed:
Handle conflicts with branch Level-9

chiayuxuan committed:
Add FindCommand class

chiayuxuan committed:
Add JavaDocs to most classes

chiayuxuan committed:
Merge pull request #1 from chiayuxuan/branch-A-JavaDoc

Merged branch-A-JavaDoc into Master

chiayuxuan committed:
Fixed bugs and added HelCommand

chiayuxuan committed:
Checked code style

chiayuxuan committed:
Updated User Guide
